jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/EtagUtil.java

```diff
@@ -0,0 +1,251 @@
+        }
+
+        Base64.Encoder encoder = Base64.getEncoder().withoutPadding();
+        long lastModifiedMillis = -1;
+        long size = -1;
+        try
+        {
+            lastModifiedMillis = Files.getLastModifiedTime(path).toMillis();
+            size = Files.size(path);
+        }
+        catch (IOException ignored)
+        {
+            // TODO: should we ignore the inability to read last modified or size?
+        }
+        b.append(encoder.encodeToString(longToBytes(lastModifiedMillis ^ lhash)));
+        b.append(encoder.encodeToString(longToBytes(size ^ lhash)));
+        if (suffix != null)
+            b.append(suffix);
+        b.append('"');
+        return b.toString();
+    }
+
+    /**
+     * Rewrite etag with a new suffix, satisfying quoting rules, and preserving optional weak flag.
+     *
+     * @param etag the original etag
+     * @param newSuffix the new suffix to add or change (if a preexisting suffix exists)
+     * @return the new etag,
+     */
+    public static String rewriteWithSuffix(String etag, String newSuffix)
+    {
+        StringBuilder ret = new StringBuilder();
+        boolean weak = etag.startsWith("W/");
+        int start = 0;
+        if (etag.startsWith("W/"))
+        {
+            weak = true;
+            start = 2;
+        }
+
+        // ignore quotes
+        while (etag.charAt(start) == '"')
+        {
+            start++;
+        }
+        int end = etag.length();
+        while (etag.charAt(end - 1) == '"')
+        {
+            end--;
+        }
+        // find suffix (if present)
+        int suffixIdx = etag.lastIndexOf('-', end);
+        if (suffixIdx >= 0 && suffixIdx >= start)
+            end = suffixIdx;
+
+        // build new etag
+        if (weak)
+            ret.append("W/");
+        ret.append('"');
+        ret.append(etag, start, end);
+        ret.append(newSuffix);
+        ret.append('"');
+        return ret.toString();
+    }
+
+    /**
+     * Test if the given Etag is considered weak.
+     *
+     * @param etag the etag to test
+     * @return true if weak.
+     */
+    public static boolean isWeak(String etag)
+    {
+        return etag.startsWith("W/");
+    }
+
+    /**
+     * Test if the given Etag is considered strong (not weak).
+     *
+     * @param etag the etag to test
+     * @return true if strong (not weak).
+     */
+    public static boolean isStrong(String etag)
+    {
+        return !isWeak(etag);
+    }
+
+    /**
+     * Test a match of an etag against an etagWithOptionalSuffix.
+     *
+     * @param etag An etag without a compression suffix
+     * @param etagWithOptionalSuffix An etag optionally with a compression suffix.
+     * @return True if the tags are equal.
+     * @see <a href="https://www.rfc-editor.org/rfc/rfc9110#section-8.8.3.2">RFC9110: Section 8.8.3.2 - Etag Comparison</a>.
+     */
+    public static boolean match(String etag, String etagWithOptionalSuffix)
```

Comment:
I would call this method `matches` because it returns a boolean.
It's kind of a query/question rather than a command.